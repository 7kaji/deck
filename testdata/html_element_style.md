# HTML Element Style Tests

---

# Title

## HTML Elements as Style Names

This is <strong>strong text</strong> and <em>emphasized text</em>.

Here is <b>bold text</b> and <i>italic text</i>.

Some <mark>marked text</mark> and <small>small text</small>.

Technical terms: <code>code text</code> and <kbd>keyboard input</kbd>.

Citations: <cite>citation text</cite> and <q>quoted text</q>.

Ruby annotation: <ruby>ruby text<rt>annotation</rt></ruby>.

---

# Mixed Usage

Text with <span class="notice">class-based style</span> and <strong>element-based style</strong>.

Nested: <strong>bold <em>and italic</em> text</strong>.

---

# Edge Cases

Empty element: <strong></strong>

Multiple elements: <b>bold</b> <i>italic</i> <u>underlined</u>.

With spaces: <strong> spaced strong </strong>.