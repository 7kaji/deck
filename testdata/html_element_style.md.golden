[
  {
    "layout": "",
    "titles": [
      "HTML Element Style Tests"
    ],
    "headings": {
      "1": [
        "HTML Element Style Tests"
      ]
    }
  },
  {
    "layout": "",
    "titles": [
      "Title"
    ],
    "subtitles": [
      "HTML Elements as Style Names"
    ],
    "bodies": [
      {
        "paragraphs": [
          {
            "fragments": [
              {
                "value": "This is "
              },
              {
                "value": "strong text",
                "style_name": "strong"
              },
              {
                "value": " and "
              },
              {
                "value": "emphasized text",
                "style_name": "em"
              },
              {
                "value": "."
              }
            ]
          },
          {
            "fragments": [
              {
                "value": "Here is "
              },
              {
                "value": "bold text",
                "style_name": "b"
              },
              {
                "value": " and "
              },
              {
                "value": "italic text",
                "style_name": "i"
              },
              {
                "value": "."
              }
            ]
          },
          {
            "fragments": [
              {
                "value": "Some "
              },
              {
                "value": "marked text",
                "style_name": "mark"
              },
              {
                "value": " and "
              },
              {
                "value": "small text",
                "style_name": "small"
              },
              {
                "value": "."
              }
            ]
          },
          {
            "fragments": [
              {
                "value": "Technical terms: "
              },
              {
                "value": "code text",
                "style_name": "code"
              },
              {
                "value": " and "
              },
              {
                "value": "keyboard input",
                "style_name": "kbd"
              },
              {
                "value": "."
              }
            ]
          },
          {
            "fragments": [
              {
                "value": "Citations: "
              },
              {
                "value": "citation text",
                "style_name": "cite"
              },
              {
                "value": " and "
              },
              {
                "value": "quoted text",
                "style_name": "q"
              },
              {
                "value": "."
              }
            ]
          },
          {
            "fragments": [
              {
                "value": "Ruby annotation: "
              },
              {
                "value": "ruby text",
                "style_name": "ruby"
              },
              {
                "value": "annotation",
                "style_name": "rt"
              },
              {
                "value": "."
              }
            ]
          }
        ]
      }
    ],
    "headings": {
      "1": [
        "Title"
      ],
      "2": [
        "HTML Elements as Style Names"
      ]
    }
  },
  {
    "layout": "",
    "titles": [
      "Mixed Usage"
    ],
    "bodies": [
      {
        "paragraphs": [
          {
            "fragments": [
              {
                "value": "Text with "
              },
              {
                "value": "class-based style",
                "style_name": "notice"
              },
              {
                "value": " and "
              },
              {
                "value": "element-based style",
                "style_name": "strong"
              },
              {
                "value": "."
              }
            ]
          },
          {
            "fragments": [
              {
                "value": "Nested: "
              },
              {
                "value": "bold ",
                "style_name": "strong"
              },
              {
                "value": "and italic",
                "style_name": "em"
              },
              {
                "value": " text"
              },
              {
                "value": "."
              }
            ]
          }
        ]
      }
    ],
    "headings": {
      "1": [
        "Mixed Usage"
      ]
    }
  },
  {
    "layout": "",
    "titles": [
      "Edge Cases"
    ],
    "bodies": [
      {
        "paragraphs": [
          {
            "fragments": [
              {
                "value": "Empty element: "
              }
            ]
          },
          {
            "fragments": [
              {
                "value": "Multiple elements: "
              },
              {
                "value": "bold",
                "style_name": "b"
              },
              {
                "value": " "
              },
              {
                "value": "italic",
                "style_name": "i"
              },
              {
                "value": " "
              },
              {
                "value": "underlined",
                "style_name": "u"
              },
              {
                "value": "."
              }
            ]
          },
          {
            "fragments": [
              {
                "value": "With spaces: "
              },
              {
                "value": " spaced strong ",
                "style_name": "strong"
              },
              {
                "value": "."
              }
            ]
          }
        ]
      }
    ],
    "headings": {
      "1": [
        "Edge Cases"
      ]
    }
  }
]